maven-core/src/main/java/org/apache/maven/project/MavenProject.java

```diff
@@ -1207,7 +1207,8 @@ private void deepCopy( MavenProject project )
         // disown the parent
 
         // copy fields
-        setFile( project.getFile() );
+        file = project.file;
+        basedir = project.basedir;
```

Comment:
Note that these assignments are actually unnecessary when called from `clone`, since the fields are already set correctly by `super.clone`. They would however be needed if called from `MavenProject.<init>(MavenProject)`; since that copy constructor is apparently neither called anywhere in Maven itself nor tested, I wonder if it even works.